android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/profile/PatientProfileViewModel.kt

```diff
@@ -74,11 +76,44 @@ constructor(
               profileViewDataMapper.transformInputToOutputModel(it) as
                 ProfileViewData.PatientProfileViewData
             // TODO only display some overflow menu items when certain conditions are met
+            refreshOverFlowMenu(healthModule = healthModule, patientProfile = it)
           }
       }
     }
   }
 
+  fun refreshOverFlowMenu(healthModule: HealthModule, patientProfile: ProfileData) {
```

Comment:
this is definitely too custom, we need to figure out a generic way to manage this